lib/rubygems/gemcutter_utilities.rb

```diff
@@ -132,12 +133,26 @@ def sign_in(sign_in_host = nil)
 
     email = ask "   Email: "
     password = ask_for_password "Password: "
+
+    hostname = Socket.gethostname
+    default_key_name = "#{hostname}"
+    key_name = ask "Key name [#{default_key_name}]: "
+    key_name = default_key_name if key_name.empty?
+
+    scope_params = {}
+
+    say "Please select scopes you want to enable for the key (y/n)"
+    API_SCOPES.each do |scope|
```

Comment:
As i've mentioned in Slack - i think asking the user to answer 7 different questions is not preferable for UX. Could we have a default set of scopes that's assumed, which the user can add/remove with provided options/arguments to `gem signin`

Reply:
> asking the user to answer 7 different questions is not preferable for UX. 

I agree that it is not as convenient as the current UX. IMHO, making the user explicitly pick scopes they want to enable when they run `gem signin` is the most secure option. I would recommend against having a default scope as it would lead to privilege escalation in the cases where user may not be using the default we pick for them. I am sure you are familiar with `the principle of least privilege`. Enabling a presumed limited default defeats the very purpose of separating scopes if we are going to group a certain number of them in the client anyway.

Reply:
I would also like to add that security and convenience are often at odds with each other. If you can think of way we could have both, it would be gold.